CIME/case/case.py

```diff
@@ -184,12 +184,13 @@ def __init__(self, case_root=None, read_only=True, record=False, non_local=False
                     machobj = Machines(machine=mach, extra_machines_dir=extra_machdir)
                 else:
                     machobj = Machines(machine=mach)
-                probed_machine = machobj.probe_machine_name()
-                if probed_machine:
-                    expect(
-                        mach == probed_machine,
-                        f"Current machine {probed_machine} does not match case machine {mach}.",
-                    )
+                if "NCAR_HOST" in os.environ:
```

Comment:
We should not be putting site-specific stuff like this in the CCS.

Reply:
I welcome suggestions for alternative solutions.

Reply:
I think the more portable way is a machine file setting that marks such a config.  But I'm fine with this going in now as is.

Reply:
@rljacob I've given some thought to how a field in the machine definition could resolve this issue, but haven't come up with anything, if you have any specific suggestions, I'm interested.